{
  "post_stream": {
    "posts": [
      {
        "id": 3345,
        "name": "Ray Jenkins",
        "username": "rjenkins",
        "avatar_template": "/user_avatar/forums.foundationdb.org/rjenkins/{size}/487_2.png",
        "created_at": "2019-02-12T22:55:17.533Z",
        "cooked": "<p>Hi Folks,</p>\n<p>Running into some problems with fdbbackup expire and looking for some guidance on debugging.</p>\n<p>Setup:<br>\n15 instance cluster running 6.0.16<br>\n3 i3.large running 2 fdbserver processes with transaction class running log<br>\n3 c5.xlarge running 4 fdbserver processes with stateless class running proxy/master/cluster controller<br>\n9 i3.xlarge running 4 fdbserver processes with storage class</p>\n<p>backup_agent is running on all nodes and fdbbackup is running in continuous mode.</p>\n<p><code>fdbbackup start -d &lt;blobstore_url&gt; -s 86400 -z</code></p>\n<p>So far so good, backups have been running fine.</p>\n<p>$ fdbbackup describe -d blobstore://xxxxxxxxxxxxxxxxs3.us-west-2.amazonaws.com/blah?bucket=some_bucket<br>\nURL: blobstore://xxxxxxxxxxxxxxxx@s3.us-west-2.amazonaws.com/blah?bucket=some_bucket<br>\nRestorable: true<br>\nSnapshot:  startVersion=680672277778  endVersion=681349765701  totalBytes=258515087913  restorable=true<br>\nSnapshot:  startVersion=681359352824  endVersion=767745402472  totalBytes=261944092554  restorable=true<br>\nSnapshot:  startVersion=767899470014  endVersion=854238123473  totalBytes=268954145233  restorable=true<br>\nSnapshot:  startVersion=854255297744  endVersion=940607774184  totalBytes=282034102462  restorable=true<br>\nSnapshot:  startVersion=940685990059  endVersion=1027047586694  totalBytes=295028242383  restorable=true<br>\nSnapshot:  startVersion=1027144233746  endVersion=1113480511478  totalBytes=307961176975  restorable=true<br>\nSnapshot:  startVersion=1113638682379  endVersion=1200023371426  totalBytes=321032543807  restorable=true<br>\nSnapshotBytes: 1995469391327<br>\nMinLogBeginVersion:      680672000491<br>\nContiguousLogEndVersion: 1241861148586<br>\nMaxLogEndVersion:        1241861148586<br>\nMinRestorableVersion:    681349765701<br>\nMaxRestorableVersion:    1241861148585</p>\n<p>In preparing to move to production I\u2019m performing some tests, including expiring backups. I began by attempting to delete backups prior to 02-06-2019 using the <code>expire_before_timestamp</code> feature.</p>\n<p><code>fdbbackup expire -d blobstore://xxxxx@s3.us-west-2.amazonaws.com/blah?bucket=some_bucket --expire_before_timestamp 2019-02-06.10:00:00 All data before version 680671939982 is deleted.</code></p>\n<p>Afterwards I inspected the bucket on S3 and no data was deleted. I then attempted to delete data before 2019-02-07.10:00:00 and received the following error. (Note: I have also tried with the --force flag as well with the same results)</p>\n<p><code>$ fdbbackup expire -d blobstore://xxxxxxxxx@s3.us-west-2.amazonaws.com/blah?bucket=some_bucket --expire_before_timestamp 2019-02-07.10:00:00 ERROR: HTTP response code not received or indicated failure Fatal Error: HTTP response code not received or indicated failure</code></p>\n<p>Performing a <code>fdbbackup describe</code> now shows some of the backups are no longer restorable, as expected, and the <code>MinLogBeginVersion</code> and <code>MinRestorableVersion</code> has increased but the size of the S3 bucket remains steady and growing due to continuous backups.</p>\n<p>$ fdbbackup describe -d blobstore://xxxxxxxxxxxxxxxx@s3.us-west-2.amazonaws.com/blah?bucket=some_bucket<br>\nURL: blobstore://xxxxxxxxx@s3.us-west-2.amazonaws.com/blah?bucket=some_bucket<br>\nRestorable: true<br>\nSnapshot:  startVersion=680672277778  endVersion=681349765701  totalBytes=258515087913  restorable=false<br>\nSnapshot:  startVersion=681359352824  endVersion=767745402472  totalBytes=261944092554  restorable=false<br>\nSnapshot:  startVersion=767899470014  endVersion=854238123473  totalBytes=268954145233  restorable=true<br>\nSnapshot:  startVersion=854255297744  endVersion=940607774184  totalBytes=282034102462  restorable=true<br>\nSnapshot:  startVersion=940685990059  endVersion=1027047586694  totalBytes=295028242383  restorable=true<br>\nSnapshot:  startVersion=1027144233746  endVersion=1113480511478  totalBytes=307961176975  restorable=true<br>\nSnapshot:  startVersion=1113638682379  endVersion=1200023371426  totalBytes=321032543807  restorable=true<br>\nSnapshotBytes: 1995469391327<br>\nMinLogBeginVersion:      767441226070<br>\nContiguousLogEndVersion: 1243110292855<br>\nMaxLogEndVersion:        1243110292855<br>\nMinRestorableVersion:    854238123473<br>\nMaxRestorableVersion:    1243110292854</p>\n<p>Additionally I can see the delete requests being received in Cloudwatch</p>\n<p><div class=\"lightbox-wrapper\"><a class=\"lightbox\" href=\"https://global.discourse-cdn.com/foundationdb/original/1X/dc9b21297880928293b7194125f1889c26a4f70a.png\" title=\"Screenshot from 2019-02-12 16-39-45.png\" rel=\"nofollow noopener\"><img src=\"https://global.discourse-cdn.com/foundationdb/optimized/1X/dc9b21297880928293b7194125f1889c26a4f70a_2_690x390.png\" alt=\"Screenshot%20from%202019-02-12%2016-39-45\" width=\"690\" height=\"390\" srcset=\"https://global.discourse-cdn.com/foundationdb/optimized/1X/dc9b21297880928293b7194125f1889c26a4f70a_2_690x390.png, https://global.discourse-cdn.com/foundationdb/optimized/1X/dc9b21297880928293b7194125f1889c26a4f70a_2_1035x585.png 1.5x, https://global.discourse-cdn.com/foundationdb/original/1X/dc9b21297880928293b7194125f1889c26a4f70a.png 2x\" data-small-upload=\"https://global.discourse-cdn.com/foundationdb/optimized/1X/dc9b21297880928293b7194125f1889c26a4f70a_2_10x10.png\"><div class=\"meta\">\n<svg class=\"fa d-icon d-icon-far-image svg-icon\" aria-hidden=\"true\"><use xlink:href=\"#far-image\"></use></svg><span class=\"filename\">Screenshot from 2019-02-12 16-39-45.png</span><span class=\"informations\">1160\u00d7656 97.3 KB</span><svg class=\"fa d-icon d-icon-discourse-expand svg-icon\" aria-hidden=\"true\"><use xlink:href=\"#discourse-expand\"></use></svg>\n</div></a></div></p>\n<p>And it appears the properties folder has been updated.</p>\n<p><div class=\"lightbox-wrapper\"><a class=\"lightbox\" href=\"https://global.discourse-cdn.com/foundationdb/original/1X/04a6a9c66fda5a5b8c3ad407f484f3162ed75470.png\" title=\"Screenshot from 2019-02-12 16-42-27.png\" rel=\"nofollow noopener\"><img src=\"https://global.discourse-cdn.com/foundationdb/optimized/1X/04a6a9c66fda5a5b8c3ad407f484f3162ed75470_2_690x173.png\" alt=\"Screenshot%20from%202019-02-12%2016-42-27\" width=\"690\" height=\"173\" srcset=\"https://global.discourse-cdn.com/foundationdb/optimized/1X/04a6a9c66fda5a5b8c3ad407f484f3162ed75470_2_690x173.png, https://global.discourse-cdn.com/foundationdb/optimized/1X/04a6a9c66fda5a5b8c3ad407f484f3162ed75470_2_1035x259.png 1.5x, https://global.discourse-cdn.com/foundationdb/optimized/1X/04a6a9c66fda5a5b8c3ad407f484f3162ed75470_2_1380x346.png 2x\" data-small-upload=\"https://global.discourse-cdn.com/foundationdb/optimized/1X/04a6a9c66fda5a5b8c3ad407f484f3162ed75470_2_10x10.png\"><div class=\"meta\">\n<svg class=\"fa d-icon d-icon-far-image svg-icon\" aria-hidden=\"true\"><use xlink:href=\"#far-image\"></use></svg><span class=\"filename\">Screenshot from 2019-02-12 16-42-27.png</span><span class=\"informations\">2534\u00d7638 44.2 KB</span><svg class=\"fa d-icon d-icon-discourse-expand svg-icon\" aria-hidden=\"true\"><use xlink:href=\"#discourse-expand\"></use></svg>\n</div></a></div></p>\n<p>It appears nothing from the <code>kvranges</code> folder has been deleted and there is no expiration date or expiration rule on the objects in the kvranges/snapshot folders. Additionally I have inspected the trace files but haven\u2019t noticed anything pertaining to my manual attempts at <code>fdbbackup expire</code>.</p>\n<p>Any tips on debugging?</p>",
        "post_number": 1,
        "post_type": 1,
        "posts_count": 19,
        "updated_at": "2019-02-12T22:58:06.725Z",
        "reply_count": 0,
        "reply_to_post_number": null,
        "quote_count": 0,
        "incoming_link_count": 353,
        "reads": 95,
        "readers_count": 94,
        "score": 1779.0,
        "yours": false,
        "topic_id": 1131,
        "topic_slug": "fdbbackup-expire-with-blobstore-returning-error-not-deleting-from-s3",
        "display_username": "Ray Jenkins",
        "primary_group_name": null,
        "flair_name": null,
        "flair_url": null,
        "flair_bg_color": null,
        "flair_color": null,
        "flair_group_id": null,
        "badges_granted": [],
        "version": 1,
        "can_edit": false,
        "can_delete": false,
        "can_recover": false,
        "can_see_hidden_post": false,
        "can_wiki": false,
        "link_counts": [
          {
            "url": "https://global.discourse-cdn.com/foundationdb/original/1X/dc9b21297880928293b7194125f1889c26a4f70a.png",
            "internal": false,
            "reflection": false,
            "title": "dc9b21297880928293b7194125f1889c26a4f70a.png",
            "clicks": 0
          },
          {
            "url": "https://global.discourse-cdn.com/foundationdb/original/1X/04a6a9c66fda5a5b8c3ad407f484f3162ed75470.png",
            "internal": false,
            "reflection": false,
            "title": "04a6a9c66fda5a5b8c3ad407f484f3162ed75470.png",
            "clicks": 0
          },
          {
            "url": "https://forums.foundationdb.org/t/backup-using-disk-snapshots/1076/9",
            "internal": true,
            "reflection": true,
            "title": "Backup using disk snapshots",
            "clicks": 0
          }
        ],
        "read": true,
        "user_title": "",
        "bookmarked": false,
        "actions_summary": [],
        "moderator": false,
        "admin": false,
        "staff": false,
        "user_id": 490,
        "hidden": false,
        "trust_level": 2,
        "deleted_at": null,
        "user_deleted": false,
        "edit_reason": null,
        "can_view_edit_history": true,
        "wiki": false,
        "post_url": "/t/fdbbackup-expire-with-blobstore-returning-error-not-deleting-from-s3/1131/1",
        "can_accept_answer": false,
        "can_unaccept_answer": false,
        "accepted_answer": false,
        "topic_accepted_answer": null,
        "can_vote": false
      },
      {
        "id": 3348,
        "name": "Steve Atherton",
        "username": "SteavedHams",
        "avatar_template": "/user_avatar/forums.foundationdb.org/steavedhams/{size}/18_2.png",
        "created_at": "2019-02-13T01:21:49.327Z",
        "cooked": "<p>There were several bugs recently fixed in backup in 6.0.18 and the prior version, including a particularly nasty HTTP request bug.  While I\u2019m not sure exactly what bad behavior you might be hitting, try these steps first:</p>\n<ol>\n<li>Upgrade to 6.0.18</li>\n<li>Delete the 3 files you see in the \u201cproperties\u201d folder.  I assume you can do this in the web UI where you took the screenshot.</li>\n<li>Run <code>fdbbackup describe</code> on your backup URL.</li>\n<li>Run <code>fdbbackup expire</code> on your backup URL as you did before.</li>\n</ol>\n<p>In case this was a point of confusion, please note that the byte counts you see in the <code>fdbbackup describe</code> output are not based on listing and summing up all of the kvrange files for each snapshot, as snapshots can contain millions of these so that could be very slow.  The byte counts are instead based on metadata files, one per snapshot, which are written after each snapshot\u2019s completion and which are deleted in expire operations after all of the kvrange files for the snapshot are deleted.</p>",
        "post_number": 2,
        "post_type": 1,
        "posts_count": 19,
        "updated_at": "2019-02-13T01:22:36.462Z",
        "reply_count": 1,
        "reply_to_post_number": null,
        "quote_count": 0,
        "incoming_link_count": 4,
        "reads": 76,
        "readers_count": 75,
        "score": 70.2,
        "yours": false,
        "topic_id": 1131,
        "topic_slug": "fdbbackup-expire-with-blobstore-returning-error-not-deleting-from-s3",
        "display_username": "Steve Atherton",
        "primary_group_name": null,
        "flair_name": null,
        "flair_url": null,
        "flair_bg_color": null,
        "flair_color": null,
        "flair_group_id": null,
        "badges_granted": [],
        "version": 1,
        "can_edit": false,
        "can_delete": false,
        "can_recover": false,
        "can_see_hidden_post": false,
        "can_wiki": false,
        "read": true,
        "user_title": "",
        "bookmarked": false,
        "actions_summary": [
          {
            "id": 2,
            "count": 2
          }
        ],
        "moderator": false,
        "admin": false,
        "staff": false,
        "user_id": 7,
        "hidden": false,
        "trust_level": 4,
        "deleted_at": null,
        "user_deleted": false,
        "edit_reason": null,
        "can_view_edit_history": true,
        "wiki": false,
        "post_url": "/t/fdbbackup-expire-with-blobstore-returning-error-not-deleting-from-s3/1131/2",
        "can_accept_answer": false,
        "can_unaccept_answer": false,
        "accepted_answer": false,
        "topic_accepted_answer": null
      },
      {
        "id": 3350,
        "name": "Ray Jenkins",
        "username": "rjenkins",
        "avatar_template": "/user_avatar/forums.foundationdb.org/rjenkins/{size}/487_2.png",
        "created_at": "2019-02-13T02:50:53.354Z",
        "cooked": "<p>Thanks <a class=\"mention\" href=\"/u/steavedhams\">@SteavedHams</a>, I will upgrade tomorrow and follow your instructions and let you know the results.</p>",
        "post_number": 3,
        "post_type": 1,
        "posts_count": 19,
        "updated_at": "2019-02-13T02:50:53.354Z",
        "reply_count": 0,
        "reply_to_post_number": 2,
        "quote_count": 0,
        "incoming_link_count": 1,
        "reads": 73,
        "readers_count": 72,
        "score": 19.6,
        "yours": false,
        "topic_id": 1131,
        "topic_slug": "fdbbackup-expire-with-blobstore-returning-error-not-deleting-from-s3",
        "display_username": "Ray Jenkins",
        "primary_group_name": null,
        "flair_name": null,
        "flair_url": null,
        "flair_bg_color": null,
        "flair_color": null,
        "flair_group_id": null,
        "badges_granted": [],
        "version": 1,
        "can_edit": false,
        "can_delete": false,
        "can_recover": false,
        "can_see_hidden_post": false,
        "can_wiki": false,
        "read": true,
        "user_title": "",
        "reply_to_user": {
          "id": 7,
          "username": "SteavedHams",
          "name": "Steve Atherton",
          "avatar_template": "/user_avatar/forums.foundationdb.org/steavedhams/{size}/18_2.png"
        },
        "bookmarked": false,
        "actions_summary": [],
        "moderator": false,
        "admin": false,
        "staff": false,
        "user_id": 490,
        "hidden": false,
        "trust_level": 2,
        "deleted_at": null,
        "user_deleted": false,
        "edit_reason": null,
        "can_view_edit_history": true,
        "wiki": false,
        "post_url": "/t/fdbbackup-expire-with-blobstore-returning-error-not-deleting-from-s3/1131/3",
        "can_accept_answer": false,
        "can_unaccept_answer": false,
        "accepted_answer": false,
        "topic_accepted_answer": null
      },
      {
        "id": 3364,
        "name": "Ray Jenkins",
        "username": "rjenkins",
        "avatar_template": "/user_avatar/forums.foundationdb.org/rjenkins/{size}/487_2.png",
        "created_at": "2019-02-13T17:48:01.001Z",
        "cooked": "<p>Hiya <a class=\"mention\" href=\"/u/steavedhams\">@SteavedHams</a> I\u2019ve rebuilt the cluster with new instances and on <code>6.0.18</code> and deleted the 3 files in the properties folder. Issuing a describe shows</p>\n<p>fdbbackup describe -d blobstore://xxxxxx@s3.us-west-2.amazonaws.com/blah?bucket=some_bucket<br>\nURL: blobstore://xxxxxxxxxx@s3.us-west-2.amazonaws.com/blah?bucket=some_bucket<br>\nRestorable: true<br>\nSnapshot:  startVersion=38380127694 (maxLogEnd -14.02 days)  endVersion=38801987336 (maxLogEnd -14.02 days)  totalBytes=343824424885  restorable=true<br>\nSnapshot:  startVersion=680672277778 (maxLogEnd -6.59 days)  endVersion=681349765701 (maxLogEnd -6.58 days)  totalBytes=258515087913  restorable=false<br>\nSnapshot:  startVersion=681359352824 (maxLogEnd -6.58 days)  endVersion=767745402472 (maxLogEnd -5.58 days)  totalBytes=261944092554  restorable=false<br>\nSnapshot:  startVersion=767899470014 (maxLogEnd -5.58 days)  endVersion=854238123473 (maxLogEnd -4.58 days)  totalBytes=268954145233  restorable=false<br>\nSnapshot:  startVersion=854255297744 (maxLogEnd -4.58 days)  endVersion=940607774184 (maxLogEnd -3.58 days)  totalBytes=282034102462  restorable=false<br>\nSnapshot:  startVersion=940685990059 (maxLogEnd -3.58 days)  endVersion=1027047586694 (maxLogEnd -2.58 days)  totalBytes=295028242383  restorable=false<br>\nSnapshot:  startVersion=1027144233746 (maxLogEnd -2.58 days)  endVersion=1113480511478 (maxLogEnd -1.58 days)  totalBytes=307961176975  restorable=false<br>\nSnapshot:  startVersion=1113638682379 (maxLogEnd -1.58 days)  endVersion=1200023371426 (maxLogEnd -0.58 days)  totalBytes=321032543807  restorable=false<br>\nSnapshotBytes: 2339293816212<br>\nMinLogBeginVersion:      38379644395 (maxLogEnd -14.02 days)<br>\nContiguousLogEndVersion: 61000828852 (maxLogEnd -13.76 days)<br>\nMaxLogEndVersion:        1249978824514 (maxLogEnd -0.00 days)<br>\nMinRestorableVersion:    38801987336 (maxLogEnd -14.02 days)<br>\nMaxRestorableVersion:    61000828851 (maxLogEnd -13.76 days)</p>\n<p>I then tried to run some expires.</p>\n<p>fdbbackup expire -d blobstore://xxxxxx@s3.us-west-2.amazonaws.com/blah?bucket=some_bucket --expire_before_timestamp 2019-02-07.10:00:00<br>\nFinal metadata update\u2026<br>\nAll data before version 0 has been deleted.</p>\n<p>Notably trying <code>--expire_before_version</code> attempts to List files and then errors</p>\n<p>$ fdbbackup expire -d blobstore://xxxxxx@s3.us-west-2.amazonaws.com/blah?bucket=some_bucket --expire_before_version 854255297744<br>\nListing files\u2026ERROR: HTTP response code not received or indicated failure<br>\nFatal Error: HTTP response code not received or indicated failure</p>\n<p>A <code>fdbbackup describe</code> now shows</p>\n<p>$ fdbbackup describe -d blobstore://xxxxxxxxxxxx@s3.us-west-2.amazonaws.com/blah?bucket=some_bucket<br>\nURL: blobstore://xxxxx@s3.us-west-2.amazonaws.com/blah?bucket=some_bucket<br>\nRestorable: true<br>\nSnapshot:  startVersion=38380127694 (maxLogEnd -14.02 days)  endVersion=38801987336 (maxLogEnd -14.02 days)  totalBytes=343824424885  restorable=false<br>\nSnapshot:  startVersion=680672277778 (maxLogEnd -6.59 days)  endVersion=681349765701 (maxLogEnd -6.58 days)  totalBytes=258515087913  restorable=false<br>\nSnapshot:  startVersion=681359352824 (maxLogEnd -6.58 days)  endVersion=767745402472 (maxLogEnd -5.58 days)  totalBytes=261944092554  restorable=false<br>\nSnapshot:  startVersion=767899470014 (maxLogEnd -5.58 days)  endVersion=854238123473 (maxLogEnd -4.58 days)  totalBytes=268954145233  restorable=false<br>\nSnapshot:  startVersion=854255297744 (maxLogEnd -4.58 days)  endVersion=940607774184 (maxLogEnd -3.58 days)  totalBytes=282034102462  restorable=true<br>\nSnapshot:  startVersion=940685990059 (maxLogEnd -3.58 days)  endVersion=1027047586694 (maxLogEnd -2.58 days)  totalBytes=295028242383  restorable=true<br>\nSnapshot:  startVersion=1027144233746 (maxLogEnd -2.58 days)  endVersion=1113480511478 (maxLogEnd -1.58 days)  totalBytes=307961176975  restorable=true<br>\nSnapshot:  startVersion=1113638682379 (maxLogEnd -1.58 days)  endVersion=1200023371426 (maxLogEnd -0.58 days)  totalBytes=321032543807  restorable=true<br>\nSnapshotBytes: 2339293816212<br>\nUnreliableEndVersion:    854247018002 (maxLogEnd -4.58 days)<br>\nMinLogBeginVersion:      854247018002 (maxLogEnd -4.58 days)<br>\nContiguousLogEndVersion: 1249978824514 (maxLogEnd -0.00 days)<br>\nMaxLogEndVersion:        1249978824514 (maxLogEnd -0.00 days)<br>\nMinRestorableVersion:    940607774184 (maxLogEnd -3.58 days)<br>\nMaxRestorableVersion:    1249978824513 (maxLogEnd -0.00 days)</p>\n<p>Inspecting the S3 bucket the size has not decreased. The properties file now contains.</p>\n<p><div class=\"lightbox-wrapper\"><a class=\"lightbox\" href=\"https://global.discourse-cdn.com/foundationdb/original/1X/b2d32670d0870f6f56ed43febea21266e1ba20bf.png\" title=\"Screenshot from 2019-02-13 11-45-37.png\" rel=\"nofollow noopener\"><img src=\"https://global.discourse-cdn.com/foundationdb/optimized/1X/b2d32670d0870f6f56ed43febea21266e1ba20bf_2_690x165.png\" alt=\"Screenshot%20from%202019-02-13%2011-45-37\" width=\"690\" height=\"165\" srcset=\"https://global.discourse-cdn.com/foundationdb/optimized/1X/b2d32670d0870f6f56ed43febea21266e1ba20bf_2_690x165.png, https://global.discourse-cdn.com/foundationdb/optimized/1X/b2d32670d0870f6f56ed43febea21266e1ba20bf_2_1035x247.png 1.5x, https://global.discourse-cdn.com/foundationdb/optimized/1X/b2d32670d0870f6f56ed43febea21266e1ba20bf_2_1380x330.png 2x\" data-small-upload=\"https://global.discourse-cdn.com/foundationdb/optimized/1X/b2d32670d0870f6f56ed43febea21266e1ba20bf_2_10x10.png\"><div class=\"meta\">\n<svg class=\"fa d-icon d-icon-far-image svg-icon\" aria-hidden=\"true\"><use xlink:href=\"#far-image\"></use></svg><span class=\"filename\">Screenshot from 2019-02-13 11-45-37.png</span><span class=\"informations\">2338\u00d7562 42.1 KB</span><svg class=\"fa d-icon d-icon-discourse-expand svg-icon\" aria-hidden=\"true\"><use xlink:href=\"#discourse-expand\"></use></svg>\n</div></a></div></p>",
        "post_number": 4,
        "post_type": 1,
        "posts_count": 19,
        "updated_at": "2019-02-13T17:48:01.001Z",
        "reply_count": 1,
        "reply_to_post_number": null,
        "quote_count": 0,
        "incoming_link_count": 7,
        "reads": 75,
        "readers_count": 74,
        "score": 55.0,
        "yours": false,
        "topic_id": 1131,
        "topic_slug": "fdbbackup-expire-with-blobstore-returning-error-not-deleting-from-s3",
        "display_username": "Ray Jenkins",
        "primary_group_name": null,
        "flair_name": null,
        "flair_url": null,
        "flair_bg_color": null,
        "flair_color": null,
        "flair_group_id": null,
        "badges_granted": [],
        "version": 1,
        "can_edit": false,
        "can_delete": false,
        "can_recover": false,
        "can_see_hidden_post": false,
        "can_wiki": false,
        "link_counts": [
          {
            "url": "https://global.discourse-cdn.com/foundationdb/original/1X/b2d32670d0870f6f56ed43febea21266e1ba20bf.png",
            "internal": false,
            "reflection": false,
            "title": "b2d32670d0870f6f56ed43febea21266e1ba20bf.png",
            "clicks": 0
          }
        ],
        "read": true,
        "user_title": "",
        "bookmarked": false,
        "actions_summary": [],
        "moderator": false,
        "admin": false,
        "staff": false,
        "user_id": 490,
        "hidden": false,
        "trust_level": 2,
        "deleted_at": null,
        "user_deleted": false,
        "edit_reason": null,
        "can_view_edit_history": true,
        "wiki": false,
        "post_url": "/t/fdbbackup-expire-with-blobstore-returning-error-not-deleting-from-s3/1131/4",
        "can_accept_answer": false,
        "can_unaccept_answer": false,
        "accepted_answer": false,
        "topic_accepted_answer": null
      },
      {
        "id": 3370,
        "name": "Steve Atherton",
        "username": "SteavedHams",
        "avatar_template": "/user_avatar/forums.foundationdb.org/steavedhams/{size}/18_2.png",
        "created_at": "2019-02-13T23:33:47.143Z",
        "cooked": "<p>This line:</p>\n<blockquote>\n<p>ContiguousLogEndVersion: 61000828852 (maxLogEnd -13.76 days)</p>\n</blockquote>\n<p>suggests that you likely hit a rare pre-6.0.18 bug where one of your log files was not successfully uploaded, breaking the continuity of the mutation stream at that point.  The logs after that point appear to be contiguous, however, so you can still restore to points in time after the next snapshot after that log discontinuity.</p>\n<p>When you began the expire to version 854255297744, the process got far enough that it marked everything prior to that version as being \u201cunreliable\u201d, meaning that files prior to that version could be missing because an expiration process began on that range.  Describe assumes that files before the Unreliable version are missing, so it will not list/scan them and they will not be included in the cached logBegin / logEnd metadata which indicates what range of log files has been verified to exist at some point in time.  (Some context here\u2026 Since backups can contain millions of files, the backup file scheme uses metadata files to indicate which snapshots were completely written and which log file ranges were verified to contiguously exist).  Your last <code>describe</code> then scanned after that point and found log continuity up to the latest log file present, shown here:</p>\n<blockquote>\n<p>ContiguousLogEndVersion: 1249978824514 (maxLogEnd -0.00 days)<br>\nMaxLogEndVersion: 1249978824514 (maxLogEnd -0.00 days)</p>\n</blockquote>\n<p>I can\u2019t explain why your expire operations seem to be unable to actually delete files and are returning errors.  Please add the <code>--log</code> flag to fdbbackup to get a trace file and look for errors containing the strings FileBackup, BackupContainer, BlobStore, or HTTP.</p>\n<p>Also, a note about using timestamps with backup operations.  This should probably be made more clear in the docs/tooling, but converting from a Timestamp to a Version requires access to the cluster from which the backup data originated.  Without that, either absolute version or relative decimal days from the last log present in the backup (which will be converted to an approximately accurate version using an assumption of 1e6 versions per second) must be used.  You specified <code>--expire_before_timestamp 2019-02-07.10:00:00</code> and the tool came up with version 0 for that timestamp.  Is it possible that the operation used a different cluster for the lookup than the source of the backup, possibly a cluster file in a default location such as ./fdb.cluster?</p>",
        "post_number": 5,
        "post_type": 1,
        "posts_count": 19,
        "updated_at": "2019-02-13T23:33:47.143Z",
        "reply_count": 0,
        "reply_to_post_number": 4,
        "quote_count": 0,
        "incoming_link_count": 2,
        "reads": 66,
        "readers_count": 65,
        "score": 23.2,
        "yours": false,
        "topic_id": 1131,
        "topic_slug": "fdbbackup-expire-with-blobstore-returning-error-not-deleting-from-s3",
        "display_username": "Steve Atherton",
        "primary_group_name": null,
        "flair_name": null,
        "flair_url": null,
        "flair_bg_color": null,
        "flair_color": null,
        "flair_group_id": null,
        "badges_granted": [],
        "version": 1,
        "can_edit": false,
        "can_delete": false,
        "can_recover": false,
        "can_see_hidden_post": false,
        "can_wiki": false,
        "read": true,
        "user_title": "",
        "reply_to_user": {
          "id": 490,
          "username": "rjenkins",
          "name": "Ray Jenkins",
          "avatar_template": "/user_avatar/forums.foundationdb.org/rjenkins/{size}/487_2.png"
        },
        "bookmarked": false,
        "actions_summary": [],
        "moderator": false,
        "admin": false,
        "staff": false,
        "user_id": 7,
        "hidden": false,
        "trust_level": 4,
        "deleted_at": null,
        "user_deleted": false,
        "edit_reason": null,
        "can_view_edit_history": true,
        "wiki": false,
        "post_url": "/t/fdbbackup-expire-with-blobstore-returning-error-not-deleting-from-s3/1131/5",
        "can_accept_answer": false,
        "can_unaccept_answer": false,
        "accepted_answer": false,
        "topic_accepted_answer": null
      },
      {
        "id": 3376,
        "name": "Ray Jenkins",
        "username": "rjenkins",
        "avatar_template": "/user_avatar/forums.foundationdb.org/rjenkins/{size}/487_2.png",
        "created_at": "2019-02-14T15:52:19.010Z",
        "cooked": "<p><a class=\"mention\" href=\"/u/steavedhams\">@SteavedHams</a> thank you for the detailed response. The <code>--log</code> option was very helpful for debugging, it turns out this was just an overlooked permissions problem on our end and after fixing expires are now working.</p>\n<p><code>&lt;Event Severity=\"20\" Time=\"1550101467.915273\" Type=\"BlobStoreEndpointRequestFailed\" ID=\"0000000000000000\" SuppressedEventCount=\"0\" ResponseCode=\"403\" ConnectionEstablished=\"1\" RemoteEndpoint=\"52.218.216.224:443:tls\" Verb=\"DELETE\" Resource=\"/some_bucket/data/foo/snapshots/snapshot,681359352824,767745402472,261944092554\" ThisTry=\"1\" Machine=\"0.0.0.0:0\" LogGroup=\"\" /&gt;</code></p>\n<p>I think it would be worthwhile to consider adding the HTTP response code to the output of fdbbackup when an error occurs or perhaps mention <code>--log</code> in the backup documentation. I will open an issue on GH.</p>\n<p>Thanks again for your help.</p>",
        "post_number": 6,
        "post_type": 1,
        "posts_count": 19,
        "updated_at": "2019-02-14T15:52:19.010Z",
        "reply_count": 1,
        "reply_to_post_number": null,
        "quote_count": 0,
        "incoming_link_count": 2,
        "reads": 67,
        "readers_count": 66,
        "score": 28.4,
        "yours": false,
        "topic_id": 1131,
        "topic_slug": "fdbbackup-expire-with-blobstore-returning-error-not-deleting-from-s3",
        "display_username": "Ray Jenkins",
        "primary_group_name": null,
        "flair_name": null,
        "flair_url": null,
        "flair_bg_color": null,
        "flair_color": null,
        "flair_group_id": null,
        "badges_granted": [],
        "version": 1,
        "can_edit": false,
        "can_delete": false,
        "can_recover": false,
        "can_see_hidden_post": false,
        "can_wiki": false,
        "read": true,
        "user_title": "",
        "bookmarked": false,
        "actions_summary": [],
        "moderator": false,
        "admin": false,
        "staff": false,
        "user_id": 490,
        "hidden": false,
        "trust_level": 2,
        "deleted_at": null,
        "user_deleted": false,
        "edit_reason": null,
        "can_view_edit_history": true,
        "wiki": false,
        "post_url": "/t/fdbbackup-expire-with-blobstore-returning-error-not-deleting-from-s3/1131/6",
        "can_accept_answer": false,
        "can_unaccept_answer": false,
        "accepted_answer": false,
        "topic_accepted_answer": null
      },
      {
        "id": 3379,
        "name": "Meng Xu",
        "username": "mengxu",
        "avatar_template": "/user_avatar/forums.foundationdb.org/mengxu/{size}/893_2.png",
        "created_at": "2019-02-14T18:22:04.071Z",
        "cooked": "<aside class=\"quote no-group\" data-username=\"rjenkins\" data-post=\"6\" data-topic=\"1131\">\n<div class=\"title\">\n<div class=\"quote-controls\"></div>\n<img loading=\"lazy\" alt=\"\" width=\"24\" height=\"24\" src=\"https://sea1.discourse-cdn.com/foundationdb/user_avatar/forums.foundationdb.org/rjenkins/48/487_2.png\" class=\"avatar\"> rjenkins:</div>\n<blockquote>\n<p>perhaps mention <code>--log</code> in the backup documentation</p>\n</blockquote>\n</aside>\n<p>Perhaps, you can also update the document with a PR on GitHub. <img src=\"https://emoji.discourse-cdn.com/twitter/slight_smile.png?v=12\" title=\":slight_smile:\" class=\"emoji\" alt=\":slight_smile:\" loading=\"lazy\" width=\"20\" height=\"20\"> It does not have to be big changes.</p>",
        "post_number": 7,
        "post_type": 1,
        "posts_count": 19,
        "updated_at": "2019-02-14T18:22:04.071Z",
        "reply_count": 0,
        "reply_to_post_number": 6,
        "quote_count": 1,
        "incoming_link_count": 1,
        "reads": 69,
        "readers_count": 68,
        "score": 18.8,
        "yours": false,
        "topic_id": 1131,
        "topic_slug": "fdbbackup-expire-with-blobstore-returning-error-not-deleting-from-s3",
        "display_username": "Meng Xu",
        "primary_group_name": null,
        "flair_name": null,
        "flair_url": null,
        "flair_bg_color": null,
        "flair_color": null,
        "flair_group_id": null,
        "badges_granted": [],
        "version": 1,
        "can_edit": false,
        "can_delete": false,
        "can_recover": false,
        "can_see_hidden_post": false,
        "can_wiki": false,
        "read": true,
        "user_title": null,
        "bookmarked": false,
        "actions_summary": [],
        "moderator": false,
        "admin": false,
        "staff": false,
        "user_id": 337,
        "hidden": false,
        "trust_level": 2,
        "deleted_at": null,
        "user_deleted": false,
        "edit_reason": null,
        "can_view_edit_history": true,
        "wiki": false,
        "post_url": "/t/fdbbackup-expire-with-blobstore-returning-error-not-deleting-from-s3/1131/7",
        "can_accept_answer": false,
        "can_unaccept_answer": false,
        "accepted_answer": false,
        "topic_accepted_answer": null
      },
      {
        "id": 3492,
        "name": "Ray Jenkins",
        "username": "rjenkins",
        "avatar_template": "/user_avatar/forums.foundationdb.org/rjenkins/{size}/487_2.png",
        "created_at": "2019-02-22T16:24:25.173Z",
        "cooked": "<p><a class=\"mention\" href=\"/u/steavedhams\">@SteavedHams</a> after resolving the expire issue we rebuilt the cluster and kicked off a new <code>fdbbackup</code>. Each night at 2:00AM PST there is a job that runs and expires any backups over 7 days old. It\u2019s currently showing the last couple days of backups as <code>restorable=false</code>.</p>\n<p>fdbbackup describe -d blobstore://xxxxxxxxxxxxx@s3.us-west-2.amazonaws.com/blah?bucket=some_bucket<br>\nURL: blobstore://xxxxxxxxxxxx@s3.us-west-2.amazonaws.com/blah?bucket=some_bucket<br>\nRestorable: true<br>\nSnapshot:  startVersion=353447944273 (maxLogEnd -2.94 days)  endVersion=353448585130 (maxLogEnd -2.94 days)  totalBytes=50749583  restorable=true<br>\nSnapshot:  startVersion=413355786099 (maxLogEnd -2.24 days)  endVersion=413356599485 (maxLogEnd -2.24 days)  totalBytes=63749211  restorable=false<br>\nSnapshot:  startVersion=413900113933 (maxLogEnd -2.24 days)  endVersion=499582938750 (maxLogEnd -1.25 days)  totalBytes=11386890817  restorable=false<br>\nSnapshot:  startVersion=499640399747 (maxLogEnd -1.25 days)  endVersion=585963699169 (maxLogEnd -0.25 days)  totalBytes=19205766899  restorable=false<br>\nSnapshotBytes: 30707156510<br>\nExpiredEndVersion:       0 (maxLogEnd -7.03 days)<br>\nUnreliableEndVersion:    0 (maxLogEnd -7.03 days)<br>\nMinLogBeginVersion:      353447805204 (maxLogEnd -2.94 days)<br>\nContiguousLogEndVersion: 353467805204 (maxLogEnd -2.94 days)<br>\nMaxLogEndVersion:        607311681429 (maxLogEnd -0.00 days)<br>\nMinRestorableVersion:    353448585130 (maxLogEnd -2.94 days)<br>\nMaxRestorableVersion:    353467805203 (maxLogEnd -2.94 days)</p>\n<p>A <code>fdbbackup status</code> shows the following.</p>\n<p>fdbbackup status<br>\nThe backup on tag `default\u2019 is restorable but continuing to blobstore://xxxxxxx@s3.us-west-2.amazonaws.com/blah?bucket=some_bucket.<br>\nSnapshot interval is 86400 seconds.  Current snapshot progress target is 25.32% (&gt;100% means the snapshot is supposed to be done)</p>\n<p>Details:<br>\nLogBytes written - 19919852683<br>\nRangeBytes written - 35519783194<br>\nLast complete log version and timestamp        - 607832819857, 02/22/19 15:59:41<br>\nLast complete snapshot version and timestamp   - 585963699169, 02/22/19 09:55:11<br>\nCurrent Snapshot start version and timestamp   - 585964113623, 02/22/19 09:55:11<br>\nExpected snapshot end version and timestamp    - 672364113623, 02/23/19 09:55:12<br>\nBackup supposed to stop at next snapshot completion - No<br>\nOlder Errors<br>\n1.82 day(s) ago : \u2018Task execution stopped due to timeout, abort, or completion by another worker\u2019 on \u2018file_backup_write_logs_5.2\u2019</p>\n<p>I am a bit confused on how to interpret the describe <code>restorable=false</code>, particularly in lieu of the status showing.</p>\n<p>Last complete log version and timestamp        - 607832819857, 02/22/19 15:59:41<br>\nLast complete snapshot version and timestamp   - 585963699169, 02/22/19 09:55:11</p>\n<p>Does this mean if I were to attempt a restore the past few days data would be lost as the Max RestorableVersion shows?</p>\n<p>MaxRestorableVersion:    353467805203 (maxLogEnd -2.94 days).</p>\n<p>If so do you have any tips on debugging or actions to take (i.e. wipe out backups and start over or should it correct itself)? I am also curious if the first expire that was executed before there was 7 days of backups might have somehow affected restorability?</p>\n<p>\u2014 snip job log \u2014<br>\ninfo expiring data before 2019-02-13 10:00:00.778124264 +0000 UTC m=-545335.225315229<br>\n[ server ] 02-20 04:00:01 ERROR - 2019/02/20 10:00:01  info expire output is All data before version 0 is deleted.</p>",
        "post_number": 8,
        "post_type": 1,
        "posts_count": 19,
        "updated_at": "2019-02-22T16:24:25.173Z",
        "reply_count": 3,
        "reply_to_post_number": null,
        "quote_count": 0,
        "incoming_link_count": 32,
        "reads": 68,
        "readers_count": 67,
        "score": 188.6,
        "yours": false,
        "topic_id": 1131,
        "topic_slug": "fdbbackup-expire-with-blobstore-returning-error-not-deleting-from-s3",
        "display_username": "Ray Jenkins",
        "primary_group_name": null,
        "flair_name": null,
        "flair_url": null,
        "flair_bg_color": null,
        "flair_color": null,
        "flair_group_id": null,
        "badges_granted": [],
        "version": 1,
        "can_edit": false,
        "can_delete": false,
        "can_recover": false,
        "can_see_hidden_post": false,
        "can_wiki": false,
        "link_counts": [
          {
            "url": "https://forums.foundationdb.org/t/aws-deployment-options/1472/7",
            "internal": true,
            "reflection": true,
            "title": "AWS Deployment Options",
            "clicks": 0
          }
        ],
        "read": true,
        "user_title": "",
        "bookmarked": false,
        "actions_summary": [],
        "moderator": false,
        "admin": false,
        "staff": false,
        "user_id": 490,
        "hidden": false,
        "trust_level": 2,
        "deleted_at": null,
        "user_deleted": false,
        "edit_reason": null,
        "can_view_edit_history": true,
        "wiki": false,
        "post_url": "/t/fdbbackup-expire-with-blobstore-returning-error-not-deleting-from-s3/1131/8",
        "can_accept_answer": false,
        "can_unaccept_answer": false,
        "accepted_answer": false,
        "topic_accepted_answer": null
      },
      {
        "id": 3494,
        "name": "Meng Xu",
        "username": "mengxu",
        "avatar_template": "/user_avatar/forums.foundationdb.org/mengxu/{size}/893_2.png",
        "created_at": "2019-02-22T18:18:02.479Z",
        "cooked": "<p><a class=\"mention\" href=\"/u/steavedhams\">@SteavedHams</a> Just a follow-up question to <a class=\"mention\" href=\"/u/rjenkins\">@rjenkins</a> question:</p>\n<p>Suppose the original backup allows  us to do point-in-time recovery.<br>\nNow we expire all backup before time t.<br>\nCan we still do the point-in-time recovery to any time after t?</p>",
        "post_number": 9,
        "post_type": 1,
        "posts_count": 19,
        "updated_at": "2019-02-22T18:18:02.479Z",
        "reply_count": 1,
        "reply_to_post_number": 8,
        "quote_count": 0,
        "incoming_link_count": 1,
        "reads": 53,
        "readers_count": 52,
        "score": 20.6,
        "yours": false,
        "topic_id": 1131,
        "topic_slug": "fdbbackup-expire-with-blobstore-returning-error-not-deleting-from-s3",
        "display_username": "Meng Xu",
        "primary_group_name": null,
        "flair_name": null,
        "flair_url": null,
        "flair_bg_color": null,
        "flair_color": null,
        "flair_group_id": null,
        "badges_granted": [],
        "version": 1,
        "can_edit": false,
        "can_delete": false,
        "can_recover": false,
        "can_see_hidden_post": false,
        "can_wiki": false,
        "read": true,
        "user_title": null,
        "reply_to_user": {
          "id": 490,
          "username": "rjenkins",
          "name": "Ray Jenkins",
          "avatar_template": "/user_avatar/forums.foundationdb.org/rjenkins/{size}/487_2.png"
        },
        "bookmarked": false,
        "actions_summary": [],
        "moderator": false,
        "admin": false,
        "staff": false,
        "user_id": 337,
        "hidden": false,
        "trust_level": 2,
        "deleted_at": null,
        "user_deleted": false,
        "edit_reason": null,
        "can_view_edit_history": true,
        "wiki": false,
        "post_url": "/t/fdbbackup-expire-with-blobstore-returning-error-not-deleting-from-s3/1131/9",
        "can_accept_answer": false,
        "can_unaccept_answer": false,
        "accepted_answer": false,
        "topic_accepted_answer": null
      },
      {
        "id": 3496,
        "name": "Steve Atherton",
        "username": "SteavedHams",
        "avatar_template": "/user_avatar/forums.foundationdb.org/steavedhams/{size}/18_2.png",
        "created_at": "2019-02-22T20:25:39.696Z",
        "cooked": "<p>You can recover to any point starting from the <em>end</em> of the first snapshot which <em>began</em> on or <em>after</em> t.</p>\n<p>To recover, you need<br>\na) A snapshot, captured over some period of time I will call T<br>\nb) Logs from the entire duration T<br>\nc) Logs after T until the point you want to restore to.</p>\n<p>So basically, when expiring it makes sense to choose snapshot start versions as expiration cutoff versions.  But for a large cluster and say a 1 week snapshot interval, rather than deleting say 3 million files every 7 days you can delete 1/7th of that each day, for example.</p>",
        "post_number": 10,
        "post_type": 1,
        "posts_count": 19,
        "updated_at": "2019-02-22T20:25:39.696Z",
        "reply_count": 0,
        "reply_to_post_number": 9,
        "quote_count": 0,
        "incoming_link_count": 0,
        "reads": 52,
        "readers_count": 51,
        "score": 10.4,
        "yours": false,
        "topic_id": 1131,
        "topic_slug": "fdbbackup-expire-with-blobstore-returning-error-not-deleting-from-s3",
        "display_username": "Steve Atherton",
        "primary_group_name": null,
        "flair_name": null,
        "flair_url": null,
        "flair_bg_color": null,
        "flair_color": null,
        "flair_group_id": null,
        "badges_granted": [],
        "version": 1,
        "can_edit": false,
        "can_delete": false,
        "can_recover": false,
        "can_see_hidden_post": false,
        "can_wiki": false,
        "read": true,
        "user_title": "",
        "reply_to_user": {
          "id": 337,
          "username": "mengxu",
          "name": "Meng Xu",
          "avatar_template": "/user_avatar/forums.foundationdb.org/mengxu/{size}/893_2.png"
        },
        "bookmarked": false,
        "actions_summary": [],
        "moderator": false,
        "admin": false,
        "staff": false,
        "user_id": 7,
        "hidden": false,
        "trust_level": 4,
        "deleted_at": null,
        "user_deleted": false,
        "edit_reason": null,
        "can_view_edit_history": true,
        "wiki": false,
        "post_url": "/t/fdbbackup-expire-with-blobstore-returning-error-not-deleting-from-s3/1131/10",
        "can_accept_answer": false,
        "can_unaccept_answer": false,
        "accepted_answer": false,
        "topic_accepted_answer": null
      },
      {
        "id": 3497,
        "name": "Steve Atherton",
        "username": "SteavedHams",
        "avatar_template": "/user_avatar/forums.foundationdb.org/steavedhams/{size}/18_2.png",
        "created_at": "2019-02-22T20:29:09.512Z",
        "cooked": "<p>Your describe output suggests that a log file is missing.  The log file would have had a begin version of 353467805203.</p>\n<p>There was a bug in FDB &lt; 6.0.18 which could fail to complete uploads successfully in the presence of certain other HTTP errors due to incorrect reuse of the TCP connection to the blob service.  It is possible that you hit this bug and that\u2019s why the log file is missing - were you running a backup agent at a version &lt; 6.0.18 around 3 days ago?</p>",
        "post_number": 11,
        "post_type": 1,
        "posts_count": 19,
        "updated_at": "2019-02-22T20:29:09.512Z",
        "reply_count": 0,
        "reply_to_post_number": 8,
        "quote_count": 0,
        "incoming_link_count": 0,
        "reads": 52,
        "readers_count": 51,
        "score": 10.4,
        "yours": false,
        "topic_id": 1131,
        "topic_slug": "fdbbackup-expire-with-blobstore-returning-error-not-deleting-from-s3",
        "display_username": "Steve Atherton",
        "primary_group_name": null,
        "flair_name": null,
        "flair_url": null,
        "flair_bg_color": null,
        "flair_color": null,
        "flair_group_id": null,
        "badges_granted": [],
        "version": 1,
        "can_edit": false,
        "can_delete": false,
        "can_recover": false,
        "can_see_hidden_post": false,
        "can_wiki": false,
        "read": true,
        "user_title": "",
        "reply_to_user": {
          "id": 490,
          "username": "rjenkins",
          "name": "Ray Jenkins",
          "avatar_template": "/user_avatar/forums.foundationdb.org/rjenkins/{size}/487_2.png"
        },
        "bookmarked": false,
        "actions_summary": [],
        "moderator": false,
        "admin": false,
        "staff": false,
        "user_id": 7,
        "hidden": false,
        "trust_level": 4,
        "deleted_at": null,
        "user_deleted": false,
        "edit_reason": null,
        "can_view_edit_history": true,
        "wiki": false,
        "post_url": "/t/fdbbackup-expire-with-blobstore-returning-error-not-deleting-from-s3/1131/11",
        "can_accept_answer": false,
        "can_unaccept_answer": false,
        "accepted_answer": false,
        "topic_accepted_answer": null
      },
      {
        "id": 3498,
        "name": "Steve Atherton",
        "username": "SteavedHams",
        "avatar_template": "/user_avatar/forums.foundationdb.org/steavedhams/{size}/18_2.png",
        "created_at": "2019-02-22T20:35:50.156Z",
        "cooked": "<p>To answer your question about restorability - Describe does a non-exhaustive search to find max restorability, it looks for a single mutation log sequence, unbroken from the oldest log file that exists in the backup.  This stops 2.94 days ago (as of the time you can the command) because of the missing log file.  But unless there are log files missing after that, you should be able to still restore to your current version, it\u2019s just that the <em>default</em> version that restore will use will not be what you want, as it will stop at the missing log file.</p>\n<p>You can verify restorability (in version 6.0.18) - which will verify via list operations each needed file in the source URL - by running</p>\n<p><code>fdbrestore start --dryrun -r &lt;url&gt; -version &lt;MaxLogEndVersion from describe output&gt;</code></p>\n<p>If this works, and you just have one missing log file (which I\u2019d still like to determine the cause of) then you can just expire data before 2.9 days ago and now your earliest log version in the backup will have version continuity up to the present version.  You can do this by<br>\n<code>fdbbackup expire -d &lt;url&gt; --delete_before_days 2.9</code></p>",
        "post_number": 12,
        "post_type": 1,
        "posts_count": 19,
        "updated_at": "2019-02-22T20:36:14.215Z",
        "reply_count": 0,
        "reply_to_post_number": 8,
        "quote_count": 0,
        "incoming_link_count": 1,
        "reads": 54,
        "readers_count": 53,
        "score": 15.8,
        "yours": false,
        "topic_id": 1131,
        "topic_slug": "fdbbackup-expire-with-blobstore-returning-error-not-deleting-from-s3",
        "display_username": "Steve Atherton",
        "primary_group_name": null,
        "flair_name": null,
        "flair_url": null,
        "flair_bg_color": null,
        "flair_color": null,
        "flair_group_id": null,
        "badges_granted": [],
        "version": 1,
        "can_edit": false,
        "can_delete": false,
        "can_recover": false,
        "can_see_hidden_post": false,
        "can_wiki": false,
        "read": true,
        "user_title": "",
        "reply_to_user": {
          "id": 490,
          "username": "rjenkins",
          "name": "Ray Jenkins",
          "avatar_template": "/user_avatar/forums.foundationdb.org/rjenkins/{size}/487_2.png"
        },
        "bookmarked": false,
        "actions_summary": [],
        "moderator": false,
        "admin": false,
        "staff": false,
        "user_id": 7,
        "hidden": false,
        "trust_level": 4,
        "deleted_at": null,
        "user_deleted": false,
        "edit_reason": null,
        "can_view_edit_history": true,
        "wiki": false,
        "post_url": "/t/fdbbackup-expire-with-blobstore-returning-error-not-deleting-from-s3/1131/12",
        "can_accept_answer": false,
        "can_unaccept_answer": false,
        "accepted_answer": false,
        "topic_accepted_answer": null
      },
      {
        "id": 3499,
        "name": "Ray Jenkins",
        "username": "rjenkins",
        "avatar_template": "/user_avatar/forums.foundationdb.org/rjenkins/{size}/487_2.png",
        "created_at": "2019-02-22T23:19:23.004Z",
        "cooked": "<p>Hi <a class=\"mention\" href=\"/u/steavedhams\">@SteavedHams</a>. I see thank you for the explanation and clarification about the \u201cdefault\u201d version. I believe all these backups have occurred on 6.0.18.</p>\n<p>$ fdbserver --version<br>\nFoundationDB 6.0 (v6.0.18)<br>\nsource version 48d84faa3e6174deb7f852ef4d314f7bad1dfa57<br>\nprotocol fdb00a570010001</p>\n<p>Thanks for the suggestions, following your instructions here is the output now, all showing restorable.</p>\n<p>$ fdbrestore start --dryrun -r <a href=\"mailto:xxxxxx@s3.us-west-2.amazonaws.com\">xxxxxx@s3.us-west-2.amazonaws.com</a>/blah?bucket=some_bucket --version 633345286610<br>\nBackup can be used to restore to version 633345286610</p>\n<p>fdbbackup expire -d blobstore://xxxxx@s3.us-west-2.amazonaws.com/blah?bucket=some_bucket --delete_before_days 3.24<br>\nFinal metadata update\u2026<br>\nAll data before 259200000000 versions (3 days) prior to latest backup log has been deleted.</p>\n<p>fdbbackup describe -d blobstore://xxxxx@s3.us-west-2.amazonaws.com/blah?bucket=some_bucket<br>\nURL: blobstore://xxxxx@s3.us-west-2.amazonaws.com/blah?bucket=some_bucket<br>\nRestorable: true<br>\nSnapshot:  startVersion=413355786099 (maxLogEnd -2.55 days)  endVersion=413356599485 (maxLogEnd -2.55 days)  totalBytes=63749211  restorable=true<br>\nSnapshot:  startVersion=413900113933 (maxLogEnd -2.54 days)  endVersion=499582938750 (maxLogEnd -1.55 days)  totalBytes=11386890817  restorable=true<br>\nSnapshot:  startVersion=499640399747 (maxLogEnd -1.55 days)  endVersion=585963699169 (maxLogEnd -0.55 days)  totalBytes=19205766899  restorable=true<br>\nSnapshotBytes: 30656406927<br>\nExpiredEndVersion:       374505286610 (maxLogEnd -3.00 days)<br>\nUnreliableEndVersion:    374505286610 (maxLogEnd -3.00 days)<br>\nMinLogBeginVersion:      413355575165 (maxLogEnd -2.55 days)<br>\nContiguousLogEndVersion: 633725286610 (maxLogEnd -0.00 days)<br>\nMaxLogEndVersion:        633725286610 (maxLogEnd -0.00 days)<br>\nMinRestorableVersion:    413356599485 (maxLogEnd -2.55 days)<br>\nMaxRestorableVersion:    633725286609 (maxLogEnd -0.00 days)</p>",
        "post_number": 13,
        "post_type": 1,
        "posts_count": 19,
        "updated_at": "2019-02-22T23:19:23.004Z",
        "reply_count": 2,
        "reply_to_post_number": null,
        "quote_count": 0,
        "incoming_link_count": 0,
        "reads": 55,
        "readers_count": 54,
        "score": 21.0,
        "yours": false,
        "topic_id": 1131,
        "topic_slug": "fdbbackup-expire-with-blobstore-returning-error-not-deleting-from-s3",
        "display_username": "Ray Jenkins",
        "primary_group_name": null,
        "flair_name": null,
        "flair_url": null,
        "flair_bg_color": null,
        "flair_color": null,
        "flair_group_id": null,
        "badges_granted": [],
        "version": 1,
        "can_edit": false,
        "can_delete": false,
        "can_recover": false,
        "can_see_hidden_post": false,
        "can_wiki": false,
        "read": true,
        "user_title": "",
        "bookmarked": false,
        "actions_summary": [],
        "moderator": false,
        "admin": false,
        "staff": false,
        "user_id": 490,
        "hidden": false,
        "trust_level": 2,
        "deleted_at": null,
        "user_deleted": false,
        "edit_reason": null,
        "can_view_edit_history": true,
        "wiki": false,
        "post_url": "/t/fdbbackup-expire-with-blobstore-returning-error-not-deleting-from-s3/1131/13",
        "can_accept_answer": false,
        "can_unaccept_answer": false,
        "accepted_answer": false,
        "topic_accepted_answer": null
      },
      {
        "id": 3500,
        "name": "Steve Atherton",
        "username": "SteavedHams",
        "avatar_template": "/user_avatar/forums.foundationdb.org/steavedhams/{size}/18_2.png",
        "created_at": "2019-02-22T23:23:23.365Z",
        "cooked": "<p>Glad you got things cleaned up, though I am a bit concerned by there being a missing log file on 6.0.18.</p>",
        "post_number": 14,
        "post_type": 1,
        "posts_count": 19,
        "updated_at": "2019-02-22T23:23:23.365Z",
        "reply_count": 0,
        "reply_to_post_number": 13,
        "quote_count": 0,
        "incoming_link_count": 0,
        "reads": 51,
        "readers_count": 50,
        "score": 10.2,
        "yours": false,
        "topic_id": 1131,
        "topic_slug": "fdbbackup-expire-with-blobstore-returning-error-not-deleting-from-s3",
        "display_username": "Steve Atherton",
        "primary_group_name": null,
        "flair_name": null,
        "flair_url": null,
        "flair_bg_color": null,
        "flair_color": null,
        "flair_group_id": null,
        "badges_granted": [],
        "version": 1,
        "can_edit": false,
        "can_delete": false,
        "can_recover": false,
        "can_see_hidden_post": false,
        "can_wiki": false,
        "read": true,
        "user_title": "",
        "reply_to_user": {
          "id": 490,
          "username": "rjenkins",
          "name": "Ray Jenkins",
          "avatar_template": "/user_avatar/forums.foundationdb.org/rjenkins/{size}/487_2.png"
        },
        "bookmarked": false,
        "actions_summary": [],
        "moderator": false,
        "admin": false,
        "staff": false,
        "user_id": 7,
        "hidden": false,
        "trust_level": 4,
        "deleted_at": null,
        "user_deleted": false,
        "edit_reason": null,
        "can_view_edit_history": true,
        "wiki": false,
        "post_url": "/t/fdbbackup-expire-with-blobstore-returning-error-not-deleting-from-s3/1131/14",
        "can_accept_answer": false,
        "can_unaccept_answer": false,
        "accepted_answer": false,
        "topic_accepted_answer": null
      },
      {
        "id": 5999,
        "name": "somasundaram",
        "username": "somu",
        "avatar_template": "https://avatars.discourse-cdn.com/v4/letter/s/ce73a5/{size}.png",
        "created_at": "2020-01-23T07:45:51.774Z",
        "cooked": "<aside class=\"quote no-group\" data-username=\"rjenkins\" data-post=\"13\" data-topic=\"1131\">\n<div class=\"title\">\n<div class=\"quote-controls\"></div>\n<img loading=\"lazy\" alt=\"\" width=\"24\" height=\"24\" src=\"https://sea1.discourse-cdn.com/foundationdb/user_avatar/forums.foundationdb.org/rjenkins/48/487_2.png\" class=\"avatar\"> rjenkins:</div>\n<blockquote>\n<p>633345286610</p>\n</blockquote>\n</aside>\n<p>Did any one tried the fdbbackup to aws s3 for the latest version of FoundationDB - 6.2.10.<br>\nI have tried the fdbbackup command as mentioned by Ray. I got the error like this<br>\n<strong>\"ERROR: Could not create backup container: Operation timed out</strong><br>\n<strong>ERROR: An error was encountered during submission</strong><br>\n<strong>Fatal Error: Backup error\"</strong></p>\n<p>Is there a particular version of foundationdb for taking backup to aws s3 bucket, if there any solutions please share.</p>",
        "post_number": 15,
        "post_type": 1,
        "posts_count": 19,
        "updated_at": "2020-01-23T07:45:51.774Z",
        "reply_count": 1,
        "reply_to_post_number": null,
        "quote_count": 1,
        "incoming_link_count": 0,
        "reads": 41,
        "readers_count": 40,
        "score": 13.2,
        "yours": false,
        "topic_id": 1131,
        "topic_slug": "fdbbackup-expire-with-blobstore-returning-error-not-deleting-from-s3",
        "display_username": "somasundaram",
        "primary_group_name": null,
        "flair_name": null,
        "flair_url": null,
        "flair_bg_color": null,
        "flair_color": null,
        "flair_group_id": null,
        "badges_granted": [],
        "version": 1,
        "can_edit": false,
        "can_delete": false,
        "can_recover": false,
        "can_see_hidden_post": false,
        "can_wiki": false,
        "read": true,
        "user_title": null,
        "bookmarked": false,
        "actions_summary": [],
        "moderator": false,
        "admin": false,
        "staff": false,
        "user_id": 634,
        "hidden": false,
        "trust_level": 1,
        "deleted_at": null,
        "user_deleted": false,
        "edit_reason": null,
        "can_view_edit_history": true,
        "wiki": false,
        "post_url": "/t/fdbbackup-expire-with-blobstore-returning-error-not-deleting-from-s3/1131/15",
        "can_accept_answer": false,
        "can_unaccept_answer": false,
        "accepted_answer": false,
        "topic_accepted_answer": null
      },
      {
        "id": 6015,
        "name": "Alex Miller",
        "username": "alexmiller",
        "avatar_template": "/user_avatar/forums.foundationdb.org/alexmiller/{size}/326_2.png",
        "created_at": "2020-01-23T23:45:44.012Z",
        "cooked": "<p>Try adding the <code>--log</code> flag to get a tracefile from fdbbackup, and either looking through that, or posting it, for more information.  An \u201cOperation timed out\u201d doesn\u2019t sound like an FDB error as much as it does something with your connection to S3?  As far as I\u2019m aware, 6.2.* should be able to back up to S3 just fine.</p>",
        "post_number": 16,
        "post_type": 1,
        "posts_count": 19,
        "updated_at": "2020-01-23T23:45:44.012Z",
        "reply_count": 0,
        "reply_to_post_number": 15,
        "quote_count": 0,
        "incoming_link_count": 0,
        "reads": 40,
        "readers_count": 39,
        "score": 8.0,
        "yours": false,
        "topic_id": 1131,
        "topic_slug": "fdbbackup-expire-with-blobstore-returning-error-not-deleting-from-s3",
        "display_username": "Alex Miller",
        "primary_group_name": null,
        "flair_name": null,
        "flair_url": null,
        "flair_bg_color": null,
        "flair_color": null,
        "flair_group_id": null,
        "badges_granted": [],
        "version": 1,
        "can_edit": false,
        "can_delete": false,
        "can_recover": false,
        "can_see_hidden_post": false,
        "can_wiki": false,
        "read": true,
        "user_title": null,
        "reply_to_user": {
          "id": 634,
          "username": "somu",
          "name": "somasundaram",
          "avatar_template": "https://avatars.discourse-cdn.com/v4/letter/s/ce73a5/{size}.png"
        },
        "bookmarked": false,
        "actions_summary": [],
        "moderator": false,
        "admin": false,
        "staff": false,
        "user_id": 13,
        "hidden": false,
        "trust_level": 4,
        "deleted_at": null,
        "user_deleted": false,
        "edit_reason": null,
        "can_view_edit_history": true,
        "wiki": false,
        "post_url": "/t/fdbbackup-expire-with-blobstore-returning-error-not-deleting-from-s3/1131/16",
        "can_accept_answer": false,
        "can_unaccept_answer": false,
        "accepted_answer": false,
        "topic_accepted_answer": null
      },
      {
        "id": 6022,
        "name": "somasundaram",
        "username": "somu",
        "avatar_template": "https://avatars.discourse-cdn.com/v4/letter/s/ce73a5/{size}.png",
        "created_at": "2020-01-24T07:16:52.201Z",
        "cooked": "<p>Thanks Alex,</p>\n<p>I have tried with the flag <strong>\u2013log</strong> in fdbbackup command, and got the trace log file,<br>\nGot the same error<br>\n<strong>\"ERROR: Could not create backup container: Operation timed out</strong><br>\n<strong>ERROR: An error was encountered during submission</strong><br>\n<strong>Fatal Error: Backup error\"</strong></p>\n<p>I have used the self signed TLS certificate and got the error in trace file <strong>certificate verification failed: unable to get local issuer certificate</strong> and <strong>BlobStoreEndpointRequestFailedRetryable</strong></p>\n<p>Is the TLS certificate should be purchased/registered from vendor.</p>\n<p><strong>fdbbackup command that i have used to take backup to s3:</strong><br>\nfdbbackup start -d blobstore://IAMUSER-accesskey:IAMUSER-secertkey@s3.us-west-1.amazonaws.com/filename?bucket=bucketname --log --tls_certificate_file /etc/foundationdb/certificate/fdb.pem --tls_ca_file /etc/foundationdb/certificate/cert.crt --tls_key_file /etc/foundationdb/certificate/private.key</p>",
        "post_number": 17,
        "post_type": 1,
        "posts_count": 19,
        "updated_at": "2020-01-24T07:16:52.201Z",
        "reply_count": 1,
        "reply_to_post_number": null,
        "quote_count": 0,
        "incoming_link_count": 1,
        "reads": 41,
        "readers_count": 40,
        "score": 18.2,
        "yours": false,
        "topic_id": 1131,
        "topic_slug": "fdbbackup-expire-with-blobstore-returning-error-not-deleting-from-s3",
        "display_username": "somasundaram",
        "primary_group_name": null,
        "flair_name": null,
        "flair_url": null,
        "flair_bg_color": null,
        "flair_color": null,
        "flair_group_id": null,
        "badges_granted": [],
        "version": 1,
        "can_edit": false,
        "can_delete": false,
        "can_recover": false,
        "can_see_hidden_post": false,
        "can_wiki": false,
        "read": true,
        "user_title": null,
        "bookmarked": false,
        "actions_summary": [],
        "moderator": false,
        "admin": false,
        "staff": false,
        "user_id": 634,
        "hidden": false,
        "trust_level": 1,
        "deleted_at": null,
        "user_deleted": false,
        "edit_reason": null,
        "can_view_edit_history": true,
        "wiki": false,
        "post_url": "/t/fdbbackup-expire-with-blobstore-returning-error-not-deleting-from-s3/1131/17",
        "can_accept_answer": false,
        "can_unaccept_answer": false,
        "accepted_answer": false,
        "topic_accepted_answer": null
      },
      {
        "id": 6023,
        "name": "Steve Atherton",
        "username": "SteavedHams",
        "avatar_template": "/user_avatar/forums.foundationdb.org/steavedhams/{size}/18_2.png",
        "created_at": "2020-01-24T07:42:04.075Z",
        "cooked": "<p>I believe this message is telling you that when it tries to validate Amazon S3\u2019s certificate, the one it presents to fdbbackup during the TLS handshake, it is unable to find that certificate\u2019s issuer locally in your CA file.</p>\n<p>If your cluster is not using TLS then you can just leave out the <code>--tls_ca_file</code> argument and fdbbackup (and other tools) should be able to find the default set of root certificates on your system somewhere, which will be the standard set of certificates accepted by browsers etc.</p>\n<p>If your cluster is using TLS (which I assume yes), then your ca_file probably contains the root certificate from which you issued certificates for your cluster processes, but you will also need to add to it the standard set of root certificates or at least the one Amazon is using.</p>\n<p>Hopefully someone else (<a class=\"mention\" href=\"/u/john_brownlee\">@john_brownlee</a> ?) can actually tell you how to do that.</p>",
        "post_number": 18,
        "post_type": 1,
        "posts_count": 19,
        "updated_at": "2020-01-24T07:42:57.879Z",
        "reply_count": 1,
        "reply_to_post_number": 17,
        "quote_count": 0,
        "incoming_link_count": 1,
        "reads": 41,
        "readers_count": 40,
        "score": 18.2,
        "yours": false,
        "topic_id": 1131,
        "topic_slug": "fdbbackup-expire-with-blobstore-returning-error-not-deleting-from-s3",
        "display_username": "Steve Atherton",
        "primary_group_name": null,
        "flair_name": null,
        "flair_url": null,
        "flair_bg_color": null,
        "flair_color": null,
        "flair_group_id": null,
        "badges_granted": [],
        "version": 1,
        "can_edit": false,
        "can_delete": false,
        "can_recover": false,
        "can_see_hidden_post": false,
        "can_wiki": false,
        "read": true,
        "user_title": "",
        "reply_to_user": {
          "id": 634,
          "username": "somu",
          "name": "somasundaram",
          "avatar_template": "https://avatars.discourse-cdn.com/v4/letter/s/ce73a5/{size}.png"
        },
        "bookmarked": false,
        "actions_summary": [],
        "moderator": false,
        "admin": false,
        "staff": false,
        "user_id": 7,
        "hidden": false,
        "trust_level": 4,
        "deleted_at": null,
        "user_deleted": false,
        "edit_reason": null,
        "can_view_edit_history": true,
        "wiki": false,
        "post_url": "/t/fdbbackup-expire-with-blobstore-returning-error-not-deleting-from-s3/1131/18",
        "can_accept_answer": false,
        "can_unaccept_answer": false,
        "accepted_answer": false,
        "topic_accepted_answer": null
      },
      {
        "id": 6025,
        "name": "somasundaram",
        "username": "somu",
        "avatar_template": "https://avatars.discourse-cdn.com/v4/letter/s/ce73a5/{size}.png",
        "created_at": "2020-01-24T13:25:04.572Z",
        "cooked": "<p>I have created the self signed TLS certificate by</p>\n<p>$ openssl req -x509 -nodes -days 365 -newkey rsa:2048 -keyout private.key -out cert.crt</p>\n<p>$ cat cert.crt private.key &gt; fdb.pem</p>\n<p>Verification of TLS self signed certificate failed</p>\n<p>$ openssl verify fdb.pem</p>\n<p>I got the error :<br>\nerror 18 at 0 depth lookup: self signed certificate<br>\nerror fdb.pem: verification failed</p>\n<p><strong>Let me know if the self signed TLS certificate creation command is OK ?</strong></p>\n<p>FDB Backup to AWS S3 bucket failed in both tls and non tls mode</p>\n<p>ERROR: Could not create backup container: Operation timed out<br>\nERROR: An error was encountered during submission<br>\nFatal Error: Backup error</p>\n<p>Thanks Steave \u2026,</p>",
        "post_number": 19,
        "post_type": 1,
        "posts_count": 19,
        "updated_at": "2020-01-24T13:25:04.572Z",
        "reply_count": 0,
        "reply_to_post_number": 18,
        "quote_count": 0,
        "incoming_link_count": 0,
        "reads": 41,
        "readers_count": 40,
        "score": 8.2,
        "yours": false,
        "topic_id": 1131,
        "topic_slug": "fdbbackup-expire-with-blobstore-returning-error-not-deleting-from-s3",
        "display_username": "somasundaram",
        "primary_group_name": null,
        "flair_name": null,
        "flair_url": null,
        "flair_bg_color": null,
        "flair_color": null,
        "flair_group_id": null,
        "badges_granted": [],
        "version": 1,
        "can_edit": false,
        "can_delete": false,
        "can_recover": false,
        "can_see_hidden_post": false,
        "can_wiki": false,
        "read": true,
        "user_title": null,
        "reply_to_user": {
          "id": 7,
          "username": "SteavedHams",
          "name": "Steve Atherton",
          "avatar_template": "/user_avatar/forums.foundationdb.org/steavedhams/{size}/18_2.png"
        },
        "bookmarked": false,
        "actions_summary": [],
        "moderator": false,
        "admin": false,
        "staff": false,
        "user_id": 634,
        "hidden": false,
        "trust_level": 1,
        "deleted_at": null,
        "user_deleted": false,
        "edit_reason": null,
        "can_view_edit_history": true,
        "wiki": false,
        "post_url": "/t/fdbbackup-expire-with-blobstore-returning-error-not-deleting-from-s3/1131/19",
        "can_accept_answer": false,
        "can_unaccept_answer": false,
        "accepted_answer": false,
        "topic_accepted_answer": null
      }
    ],
    "stream": [
      3345,
      3348,
      3350,
      3364,
      3370,
      3376,
      3379,
      3492,
      3494,
      3496,
      3497,
      3498,
      3499,
      3500,
      5999,
      6015,
      6022,
      6023,
      6025
    ]
  },
  "timeline_lookup": [
    [
      1,
      2443
    ],
    [
      4,
      2442
    ],
    [
      6,
      2441
    ],
    [
      8,
      2433
    ],
    [
      15,
      2099
    ],
    [
      16,
      2098
    ],
    [
      19,
      2097
    ]
  ],
  "suggested_topics": [],
  "tags": [],
  "tags_descriptions": {},
  "fancy_title": "Fdbbackup expire with blobstore returning error, not deleting from S3",
  "id": 1131,
  "title": "Fdbbackup expire with blobstore returning error, not deleting from S3",
  "posts_count": 19,
  "created_at": "2019-02-12T22:55:17.346Z",
  "views": 2227,
  "reply_count": 11,
  "like_count": 2,
  "last_posted_at": "2020-01-24T13:25:04.572Z",
  "visible": true,
  "closed": false,
  "archived": false,
  "has_summary": false,
  "archetype": "regular",
  "slug": "fdbbackup-expire-with-blobstore-returning-error-not-deleting-from-s3",
  "category_id": 7,
  "word_count": 3577,
  "deleted_at": null,
  "user_id": 490,
  "featured_link": null,
  "pinned_globally": false,
  "pinned_at": null,
  "pinned_until": null,
  "image_url": "https://global.discourse-cdn.com/foundationdb/optimized/1X/dc9b21297880928293b7194125f1889c26a4f70a_2_1024x579.png",
  "slow_mode_seconds": 0,
  "draft": null,
  "draft_key": "topic_1131",
  "draft_sequence": null,
  "unpinned": null,
  "pinned": false,
  "current_post_number": 1,
  "highest_post_number": 19,
  "deleted_by": null,
  "actions_summary": [
    {
      "id": 4,
      "count": 0,
      "hidden": false,
      "can_act": false
    },
    {
      "id": 8,
      "count": 0,
      "hidden": false,
      "can_act": false
    },
    {
      "id": 10,
      "count": 0,
      "hidden": false,
      "can_act": false
    },
    {
      "id": 7,
      "count": 0,
      "hidden": false,
      "can_act": false
    }
  ],
  "chunk_size": 20,
  "bookmarked": false,
  "topic_timer": null,
  "message_bus_last_id": 0,
  "participant_count": 5,
  "show_read_indicator": false,
  "thumbnails": [
    {
      "max_width": null,
      "max_height": null,
      "width": 1160,
      "height": 656,
      "url": "https://global.discourse-cdn.com/foundationdb/original/1X/dc9b21297880928293b7194125f1889c26a4f70a.png"
    },
    {
      "max_width": 1024,
      "max_height": 1024,
      "width": 1024,
      "height": 579,
      "url": "https://global.discourse-cdn.com/foundationdb/optimized/1X/dc9b21297880928293b7194125f1889c26a4f70a_2_1024x579.png"
    }
  ],
  "slow_mode_enabled_until": null,
  "tags_disable_ads": false,
  "related_topics": [
    {
      "fancy_title": "Sanity check: FoundationDB backup blobstore credentials permissions errors",
      "id": 1944,
      "title": "Sanity check: FoundationDB backup blobstore credentials permissions errors",
      "slug": "sanity-check-foundationdb-backup-blobstore-credentials-permissions-errors",
      "posts_count": 20,
      "reply_count": 4,
      "highest_post_number": 20,
      "image_url": null,
      "created_at": "2020-02-13T17:22:53.536Z",
      "last_posted_at": "2020-02-19T00:40:20.641Z",
      "bumped": true,
      "bumped_at": "2020-02-19T00:40:20.641Z",
      "archetype": "regular",
      "unseen": false,
      "pinned": false,
      "unpinned": null,
      "visible": true,
      "closed": false,
      "archived": false,
      "bookmarked": null,
      "liked": null,
      "tags": [],
      "tags_descriptions": {},
      "like_count": 1,
      "views": 1493,
      "category_id": 7,
      "featured_link": null,
      "has_accepted_answer": false,
      "posters": [
        {
          "extras": null,
          "description": "Original Poster",
          "user": {
            "id": 484,
            "username": "ksnavely",
            "name": "Kyle Snavely",
            "avatar_template": "/user_avatar/forums.foundationdb.org/ksnavely/{size}/483_2.png",
            "trust_level": 1
          }
        },
        {
          "extras": null,
          "description": "Frequent Poster",
          "user": {
            "id": 13,
            "username": "alexmiller",
            "name": "Alex Miller",
            "avatar_template": "/user_avatar/forums.foundationdb.org/alexmiller/{size}/326_2.png",
            "trust_level": 4
          }
        },
        {
          "extras": "latest",
          "description": "Most Recent Poster",
          "user": {
            "id": 7,
            "username": "SteavedHams",
            "name": "Steve Atherton",
            "avatar_template": "/user_avatar/forums.foundationdb.org/steavedhams/{size}/18_2.png",
            "trust_level": 4
          }
        }
      ]
    },
    {
      "fancy_title": "FoundationDB fdbbackup",
      "id": 1460,
      "title": "FoundationDB fdbbackup",
      "slug": "foundationdb-fdbbackup",
      "posts_count": 7,
      "reply_count": 2,
      "highest_post_number": 7,
      "image_url": null,
      "created_at": "2019-06-17T18:54:47.516Z",
      "last_posted_at": "2019-10-09T03:07:15.974Z",
      "bumped": true,
      "bumped_at": "2019-10-09T03:07:15.974Z",
      "archetype": "regular",
      "unseen": false,
      "pinned": false,
      "unpinned": null,
      "visible": true,
      "closed": false,
      "archived": false,
      "bookmarked": null,
      "liked": null,
      "tags": [],
      "tags_descriptions": {},
      "like_count": 1,
      "views": 2461,
      "category_id": 7,
      "featured_link": null,
      "has_accepted_answer": false,
      "posters": [
        {
          "extras": null,
          "description": "Original Poster",
          "user": {
            "id": 579,
            "username": "approachcp",
            "name": "",
            "avatar_template": "https://avatars.discourse-cdn.com/v4/letter/a/f04885/{size}.png",
            "trust_level": 1
          }
        },
        {
          "extras": null,
          "description": "Frequent Poster",
          "user": {
            "id": 581,
            "username": "surprisetalk",
            "name": "Taylor Sarrafian",
            "avatar_template": "/user_avatar/forums.foundationdb.org/surprisetalk/{size}/579_2.png",
            "trust_level": 1
          }
        },
        {
          "extras": "latest",
          "description": "Most Recent Poster",
          "user": {
            "id": 7,
            "username": "SteavedHams",
            "name": "Steve Atherton",
            "avatar_template": "/user_avatar/forums.foundationdb.org/steavedhams/{size}/18_2.png",
            "trust_level": 4
          }
        }
      ]
    },
    {
      "fancy_title": "&ldquo;backup failure: Could not create backup container: Operation timed out&rdquo; with version 6.3.9",
      "id": 2703,
      "title": "\"backup failure: Could not create backup container: Operation timed out\" with version 6.3.9",
      "slug": "backup-failure-could-not-create-backup-container-operation-timed-out-with-version-6-3-9",
      "posts_count": 7,
      "reply_count": 1,
      "highest_post_number": 7,
      "image_url": null,
      "created_at": "2021-05-17T11:07:44.747Z",
      "last_posted_at": "2022-01-05T11:17:46.398Z",
      "bumped": true,
      "bumped_at": "2022-01-05T15:04:16.923Z",
      "archetype": "regular",
      "unseen": false,
      "pinned": false,
      "unpinned": null,
      "visible": true,
      "closed": false,
      "archived": false,
      "bookmarked": null,
      "liked": null,
      "tags": [],
      "tags_descriptions": {},
      "like_count": 0,
      "views": 1224,
      "category_id": 7,
      "featured_link": null,
      "has_accepted_answer": false,
      "posters": [
        {
          "extras": null,
          "description": "Original Poster",
          "user": {
            "id": 865,
            "username": "manish-2014",
            "name": "Manish Sharan",
            "avatar_template": "/user_avatar/forums.foundationdb.org/manish-2014/{size}/1002_2.png",
            "trust_level": 1
          }
        },
        {
          "extras": null,
          "description": "Frequent Poster",
          "user": {
            "id": 14,
            "username": "andrew.noyes",
            "name": "",
            "avatar_template": "/user_avatar/forums.foundationdb.org/andrew.noyes/{size}/443_2.png",
            "trust_level": 2
          }
        },
        {
          "extras": null,
          "description": "Frequent Poster",
          "user": {
            "id": 13,
            "username": "alexmiller",
            "name": "Alex Miller",
            "avatar_template": "/user_avatar/forums.foundationdb.org/alexmiller/{size}/326_2.png",
            "trust_level": 4
          }
        },
        {
          "extras": "latest",
          "description": "Most Recent Poster",
          "user": {
            "id": 949,
            "username": "DenisBY",
            "name": "Denis",
            "avatar_template": "/user_avatar/forums.foundationdb.org/denisby/{size}/1104_2.png",
            "trust_level": 1
          }
        }
      ]
    },
    {
      "fancy_title": "Fdbbackup to blobstore failed using blob_credentials",
      "id": 2794,
      "title": "Fdbbackup to blobstore failed using blob_credentials",
      "slug": "fdbbackup-to-blobstore-failed-using-blob-credentials",
      "posts_count": 1,
      "reply_count": 0,
      "highest_post_number": 1,
      "image_url": null,
      "created_at": "2021-07-16T02:42:49.849Z",
      "last_posted_at": "2021-07-16T02:42:49.932Z",
      "bumped": true,
      "bumped_at": "2021-07-16T06:52:52.308Z",
      "archetype": "regular",
      "unseen": false,
      "pinned": false,
      "unpinned": null,
      "visible": true,
      "closed": false,
      "archived": false,
      "bookmarked": null,
      "liked": null,
      "tags": [],
      "tags_descriptions": {},
      "like_count": 0,
      "views": 481,
      "category_id": 7,
      "featured_link": null,
      "has_accepted_answer": false,
      "posters": [
        {
          "extras": "latest single",
          "description": "Original Poster, Most Recent Poster",
          "user": {
            "id": 965,
            "username": "boorishwreck",
            "name": "",
            "avatar_template": "/user_avatar/forums.foundationdb.org/boorishwreck/{size}/1135_2.png",
            "trust_level": 1
          }
        }
      ]
    },
    {
      "fancy_title": "Trying to restore a backup from S3",
      "id": 2502,
      "title": "Trying to restore a backup from S3",
      "slug": "trying-to-restore-a-backup-from-s3",
      "posts_count": 3,
      "reply_count": 0,
      "highest_post_number": 3,
      "image_url": null,
      "created_at": "2020-12-29T14:02:35.489Z",
      "last_posted_at": "2021-01-05T12:58:17.831Z",
      "bumped": true,
      "bumped_at": "2021-01-05T12:58:17.831Z",
      "archetype": "regular",
      "unseen": false,
      "pinned": false,
      "unpinned": null,
      "visible": true,
      "closed": false,
      "archived": false,
      "bookmarked": null,
      "liked": null,
      "tags": [],
      "tags_descriptions": {},
      "like_count": 0,
      "views": 674,
      "category_id": 7,
      "featured_link": null,
      "has_accepted_answer": false,
      "posters": [
        {
          "extras": "latest",
          "description": "Original Poster, Most Recent Poster",
          "user": {
            "id": 872,
            "username": "COran",
            "name": "Carlito",
            "avatar_template": "/user_avatar/forums.foundationdb.org/coran/{size}/1006_2.png",
            "trust_level": 1
          }
        },
        {
          "extras": null,
          "description": "Frequent Poster",
          "user": {
            "id": 7,
            "username": "SteavedHams",
            "name": "Steve Atherton",
            "avatar_template": "/user_avatar/forums.foundationdb.org/steavedhams/{size}/18_2.png",
            "trust_level": 4
          }
        }
      ]
    }
  ],
  "summarizable": false,
  "can_vote": false,
  "vote_count": 0,
  "user_voted": false,
  "discourse_zendesk_plugin_zendesk_id": null,
  "discourse_zendesk_plugin_zendesk_url": "https://your-url.zendesk.com/agent/tickets/",
  "details": {
    "can_edit": false,
    "notification_level": 1,
    "participants": [
      {
        "id": 7,
        "username": "SteavedHams",
        "name": "Steve Atherton",
        "avatar_template": "/user_avatar/forums.foundationdb.org/steavedhams/{size}/18_2.png",
        "post_count": 7,
        "primary_group_name": null,
        "flair_name": null,
        "flair_url": null,
        "flair_color": null,
        "flair_bg_color": null,
        "flair_group_id": null,
        "trust_level": 4
      },
      {
        "id": 490,
        "username": "rjenkins",
        "name": "Ray Jenkins",
        "avatar_template": "/user_avatar/forums.foundationdb.org/rjenkins/{size}/487_2.png",
        "post_count": 6,
        "primary_group_name": null,
        "flair_name": null,
        "flair_url": null,
        "flair_color": null,
        "flair_bg_color": null,
        "flair_group_id": null,
        "trust_level": 2
      },
      {
        "id": 634,
        "username": "somu",
        "name": "somasundaram",
        "avatar_template": "https://avatars.discourse-cdn.com/v4/letter/s/ce73a5/{size}.png",
        "post_count": 3,
        "primary_group_name": null,
        "flair_name": null,
        "flair_url": null,
        "flair_color": null,
        "flair_bg_color": null,
        "flair_group_id": null,
        "trust_level": 1
      },
      {
        "id": 337,
        "username": "mengxu",
        "name": "Meng Xu",
        "avatar_template": "/user_avatar/forums.foundationdb.org/mengxu/{size}/893_2.png",
        "post_count": 2,
        "primary_group_name": null,
        "flair_name": null,
        "flair_url": null,
        "flair_color": null,
        "flair_bg_color": null,
        "flair_group_id": null,
        "trust_level": 2
      },
      {
        "id": 13,
        "username": "alexmiller",
        "name": "Alex Miller",
        "avatar_template": "/user_avatar/forums.foundationdb.org/alexmiller/{size}/326_2.png",
        "post_count": 1,
        "primary_group_name": null,
        "flair_name": null,
        "flair_url": null,
        "flair_color": null,
        "flair_bg_color": null,
        "flair_group_id": null,
        "trust_level": 4
      }
    ],
    "created_by": {
      "id": 490,
      "username": "rjenkins",
      "name": "Ray Jenkins",
      "avatar_template": "/user_avatar/forums.foundationdb.org/rjenkins/{size}/487_2.png"
    },
    "last_poster": {
      "id": 634,
      "username": "somu",
      "name": "somasundaram",
      "avatar_template": "https://avatars.discourse-cdn.com/v4/letter/s/ce73a5/{size}.png"
    }
  },
  "bookmarks": []
}